src/idl_parser.cpp

```diff
@@ -878,11 +878,11 @@ CheckedError Parser::ParseAnyValue(Value &val, FieldDef *field,
           (token_ == kTokenIdentifier || token_ == kTokenStringConstant)) {
         ECHECK(ParseHash(val, field));
       } else {
-        ECHECK(ParseSingleValue(val));
+        ECHECK(ParseSingleValue(field ? field->name : "", val));
```

Comment:
This is going to allocate a std::string for every number parsed in a vector. Why don't you make the argument to `ParseSingleValue` a `*` so the the null-check can happen only when the error occurs?